src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -35,17 +35,15 @@ public static VisualStudioInstance GetInstance(string dotNetSdkPath)
                 return null;
             }
 
-            string versionContent = File.ReadAllText(versionPath);
-            Match versionMatch = VersionRegex.Match(versionContent);
+            // Preview versions contain a hyphen after the numeric part of the version. Version.TryParse doesn't accept that.
+            Match versionMatch = VersionRegex.Match(File.ReadAllText(versionPath));
 
-            if (!versionMatch.Success)
-            {
-                return null;
-            }
-
-            if (!int.TryParse(versionMatch.Groups[1].Value, out int major) ||
+            if (!versionMatch.Success ||
+                !int.TryParse(versionMatch.Groups[1].Value, out int major) ||
                 !int.TryParse(versionMatch.Groups[2].Value, out int minor) ||
-                !int.TryParse(versionMatch.Groups[3].Value, out int patch))
+                !int.TryParse(versionMatch.Groups[3].Value, out int patch) ||
+                major > Environment.Version.Major ||
+                (major == Environment.Version.Major && minor > Environment.Version.Minor))
```

Comment:
This definitely needs a comment! Describe all the things you've learned about why we care about the current runtime version relative to the SDK version, and the SDK policy as we currently understand it.